rust/main/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -611,32 +611,21 @@ where
                 )
             })
             .collect::<ChainResult<Vec<_>>>()?;
-        if let Some(contract_call) = contract_calls.first() {
-            self.refresh_block_and_fee_cache(&contract_call.tx).await?;
-        }
-        let filled_tx_params_futures = contract_calls.iter().map(|tx| {
-            fill_tx_gas_params(
-                tx.clone(),
-                self.provider.clone(),
-                &self.conn.transaction_overrides,
-                &self.domain,
-                true,
-                self.cache.clone(),
-            )
-        });
-        let contract_calls = join_all(filled_tx_params_futures)
-            .await
-            .into_iter()
-            .collect::<ChainResult<Vec<_>>>()?;
 
-        if self.conn.operation_batch.bypass_batch_simulation {
-            // submit the tx without checking if subcalls would revert
-            self.submit_multicall(&mut multicall, contract_calls, self.cache.clone())
-                .await
-        } else {
-            self.simulate_and_submit_batch(&mut multicall, contract_calls, self.cache.clone())
-                .await
-        }
+        let simulate_future = self.simulate_batch(&mut multicall, contract_calls.clone());
+        let refresh_cache_future = async {
+            if let Some(contract_call) = contract_calls.first() {
+                self.refresh_block_and_fee_cache(&contract_call.tx).await
+            } else {
+                Ok(())
+            }
+        };
+
+        let (simulate_result, refresh_result) = join!(simulate_future, refresh_cache_future);
+
+        let (simulation, _) = (simulate_result?, refresh_result?);
```

Comment:
we shouldn't propagate the error if `refresh_result` fails - it's fine to just not have the cache

Reply:
It is implemented in the next PR - https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/5974